go-controller/pkg/clustermanager/clustermanager.go

```diff
@@ -0,0 +1,220 @@
+package clustermanager
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+// ClusterManager structure is the object which manages the cluster nodes.
+// It creates a default network controller for the default network and a
+// secondary network cluster controller manager to manage the multi networks.
+type ClusterManager struct {
+	client                      clientset.Interface
+	defaultNetClusterController *networkClusterController
+	wf                          *factory.WatchFactory
+	wg                          *sync.WaitGroup
+	secondaryNetClusterManager  *secondaryNetworkClusterManager
+	// event recorder used to post events to k8s
+	recorder record.EventRecorder
+
+	// unique identity for clusterManager running on different ovnkube-cluster-manager instance,
+	// used for leader election
+	identity string
+
+	isActive   bool //Indicates if cluster manager is leader and it is active
```

Comment:
nit: nothing against having this flag. But hopefully not hiding something that should be idempotent but it isn't. Might be worth to add a comment what is the thing that should not be called twice that this is protecting against.

Reply:
Ack.  I'll add comments.